src/MongoDB.Driver.Core/Core/Bindings/IChannel.cs

```diff
@@ -118,9 +121,11 @@ public interface IChannel : IDisposable
             ReadPreference readPreference,
             DatabaseNamespace databaseNamespace,
             BsonDocument command,
+            IEnumerable<Type1CommandMessageSection> commandPayloads,
             IElementNameValidator commandValidator,
             BsonDocument additionalOptions,
-            Func<CommandResponseHandling> responseHandling,
+            Action<IMessageEncoderPostProcessor> postWriteAction,
+            CommandResponseHandling responseHandling,
```

Comment:
We used to postpone deciding how to handle the response by passing in a Func.

Now we just pass in the expected response handling. In *some* cases the postWriteAction will result in a different response handling: when w0 is changed to w1 response handling changes from NoneExpected to Return.